nifi-nar-bundles/nifi-standard-services/nifi-hbase_1_1_2-client-service-bundle/nifi-hbase_1_1_2-client-service/src/main/java/org/apache/nifi/hbase/HBase_1_1_2_ClientService.java

```diff
@@ -336,51 +346,85 @@ public void shutdown() {
         }
     }
 
+    private static final byte[] EMPTY_VIS_STRING;
+
+    static {
+        try {
+            EMPTY_VIS_STRING = "".getBytes("UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private List<Put> buildPuts(byte[] rowKey, List<PutColumn> columns) {
+        List<Put> retVal = new ArrayList<>();
+
+        try {
+            Put put = null;
+
+            for (final PutColumn column : columns) {
```

Comment:
Allowing the different columns in same row to have diff visibility right.  For each of the unique visibility one Put need to be there along with corresponding columns in it.  I believe the below logic is doing that. Just saying.

Reply:
In the list of PutColumn I can have 1st column having 'x' as visibility and next one with 'y' and 3rd one again with 'x'. So here you will create 3 Put objects I guess. Would be better if we map 'columns' into a visibility vs columns data structure and then for each of visibility create one Put (?)

Reply:
@anoopsjohn Good point. I think you're correct. Bundling columns into the same put would be able to reduce the number of Put objects. I wonder how much performance gain we can get by doing so. If you are interested in making such optimization, please go ahead and create another PR. Before/after benchmarking result would be appreciated. Thanks!